data-serving/scripts/convert-data/parsers.py

```diff
@@ -191,52 +191,42 @@ def parse_location(value: Any) -> Dict[str, Any]:
           'administrativeAreaLevel2': str,
           'locality': str,
           'geometry': {
-          'latitude': float,
-          'longitude': float
+            'latitude': float,
+            'longitude': float
           }
         }
     '''
     if pd.isna(value) or value.lower() in ['no', 'none', 'unknown']:
-        return None
+        raise ValueError('boolean value is not a valid location')
     if type(value) is not str:
         raise ValueError('location is not a string')
 
-    # Remove common prefixes 'travelled to' and 'traveled to' and lowercase it.
-    normalized_location = re.sub(
-        r'(travelled|traveled)\sto\s(the\s*)?',
-        '', value.lower())
+    # Parse the location string into tokens (e.g. city, province, country) and
+    # try to find a match with the geocoder.
+    geocode_result = lookup_location(geocoder, parse_list(value.lower(), ','))
 
-    # First try to look up the location in a hard-coded map that accounts for
-    # most popular locations that the lookup (below) misses.
-    location = LOCATIONS.get(normalized_location)
-    if location is not None:
-        return location
+    if geocode_result is None:
+        raise ValueError('no geocode found for location')
```

Comment:
The geocode functions raise `ValueError` if more than one result is found, and return `None` if less than one result is found, which then gets converted here into `ValueError`. Can these paths be simplified to "raise `ValueError` if there is not exactly one geocode match"?

Reply:
yeah fair -- i tried to simplify quite a bit by note using None at all -- just ValueErrors (and tried to centralize that logic). LMK what you think